resource/resourceadapters/limiter_test.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resourceadapters_test
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/v2"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/resource/resourceadapters"
+	coretesting "github.com/juju/juju/testing"
+)
+
+type LimiterSuite struct {
+}
+
+var _ = gc.Suite(&LimiterSuite{})
+
+var shortAttempt = &utils.AttemptStrategy{
+	Total: coretesting.ShortWait,
+	Delay: 10 * time.Millisecond,
+}
+
+func (s *LimiterSuite) TestNoLimits(c *gc.C) {
+
+	const totalToAcquire = 10
+	limiter := resourceadapters.NewResourceDownloadLimiter(0, 0)
+
+	totalAcquiredCount := int32(0)
+	trigger := make(chan struct{})
+	finished := sync.WaitGroup{}
+	for i := 0; i < totalToAcquire; i++ {
+		finished.Add(1)
+		go func() {
+			defer finished.Done()
+			limiter.Acquire("app1")
```

Comment:
before this, add synchronisation to ensure all goroutines have started first, we essentially want them all to start the race at the same time